sqlite_query.c

```diff
@@ -145,8 +152,12 @@ sqlite_convert_to_pg(Form_pg_attribute att, sqlite3_stmt * stmt, int stmt_colid,
 				{
 					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
 					{
-						int			value = sqlite3_column_int(stmt, stmt_colid);
-						return (struct NullableDatum) {Int32GetDatum(value), false};
+						sqlite_int64 i64v = sqlite3_column_int64(stmt, stmt_colid);
+						if (i64v > PG_INT32_MAX || i64v < PG_INT32_MIN)
```

Comment:
What about int64 overflow? It happens when we insert too big number (PG_INT64_MAX + 1) to SQLite column, SQLite will store it as a double value: 

> If the TEXT value is a well-formed integer literal that is too large to fit in a 64-bit signed integer, it is converted to REAL. https://www.sqlite.org/datatype3.html

I think we can check simply by
```C
case INT8OID:
{
		switch (sqlite_value_affinity)
		{
				....
				case SQLITE_DOUBLE:
				{
				        double		value = sqlite3_column_double(stmt, stmt_colid);
				        Datum     d = DirectFunctionCall1(dtoi8, Float8GetDatum((float8) value));
				        return (struct NullableDatum) {d, false};
				}
....
```

Reply:
Added in https://github.com/pgspider/sqlite_fdw/pull/91/commits/06ee232ebf1785ea5daed0391f5aa12d890b6dd6 without testing. `float` data for `bigint`/`int8` PostgreSQL column is not normal.

Reply:
Thank you.

Could you add test for this code. Leaving a new source code untested is not a good practice.

Reply:
Hm... This code is not a good practice at all  ;-) because we can lost not integer part of the value. I'll add overflow TC in `out_of_range` tests, but it's better to think how we can exclude this `INT8OID` + `SQLITE_DOUBLE` branch needed only in SQLite `mod()` context.

Reply:
Here is no overflow, @nxhai98. There is very bad behaviour. What should I test here?
```
ALTER FOREIGN TABLE INT8_TBL ALTER COLUMN q1 TYPE double precision;
INSERT INTO INT8_TBL VALUES (-9223372036854775809, 0); // https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC
+ALTER FOREIGN TABLE INT8_TBL ALTER COLUMN q1 TYPE int8;
+SELECT * FROM INT8_TBL;
+          q1          |        q2         
+----------------------+-------------------
+ -9223372036854775808 |                 0
```
We have -922337203685477580**9** in `INSERT`, but  -922337203685477580**8** in `SELECT` and no out or range error.

Reply:
It's because of floating point precision, `-9223372036854775808` and `-9223372036854775809` have same binary presentation in float type => the error will not happen. If we use `PG_INT64_MAX + 1` and higher or much lower than PG_INT64_MIN like `10 * PG_INT64_MIN`, the out of range error will be happen.

Reply:
Thanks for clarification, @nxhai98 , changed to -9223372036854775810 in https://github.com/pgspider/sqlite_fdw/pull/91/commits/c79a5281848396153a582386e9fd79ac7678d855 and no problems.

Reply:
Also added `10 * PG_INT64_MIN`  TCs in https://github.com/pgspider/sqlite_fdw/pull/91/commits/ddd6a141af7ecf2db9a38ae19c97f270bfe4689c and leaved TCs with problematic behaviour between `PG_INT64_MIN` and `10 * PG_INT64_MIN`.

Reply:
Thank you. Please help me resolve this thread.